src/ext.rs

```diff
@@ -0,0 +1,51 @@
+//! HTTP extensions
+
+use bytes::Bytes;
+#[cfg(feature = "http1")]
+use http::header::{GetAll, HeaderName, IntoHeaderName};
+use http::HeaderMap;
+
+/// A map from header names to their original casing as received in an HTTP response.
```

Comment:
This extension is used for both requests and responses.

Maybe after a brief description of what it is, would it make sense to have request and response examples? And/or maybe how it's used for clients and servers? Even more docs isn't required for merging, just rambling here...